### PR TITLE
.bit to Torus

### PR DESCRIPTION
**Pull request by bigbigkent**
add .bit support when transfer

### CODE REVIEW

File: src/utils/enums.js [hunk 1]
```diff
@@ -169,6 +169,8 @@ export const AVALANCHE_MAINNET_BLOCK_EXPLORER = 'https://snowtrace.io'
 export const AVALANCHE_TESTNET_URL = 'https://api.avax-test.network/ext/bc/C/rpc'
 export const AVALANCHE_TESTNET_BLOCK_EXPLORER = 'https://testnet.snowtrace.io'
 
+export const BIT_HOST_URL = 'https://indexer-basic.did.id'
```
Comment (zgayjjf) [inline src/utils/enums.js:172]: user `indexer-v1`

File: src/containers/WalletTransfer/WalletTransfer.vue [hunk 2]
```diff
@@ -793,6 +820,8 @@ export default {
         this.selectedVerifier = ENS
       } else if (/.crypto$/.test(toAddress)) {
         this.selectedVerifier = UNSTOPPABLE_DOMAINS
+      } else if (toAddress.indexOf(this.bitTail) > 0 && toAddress.indexOf(this.bitTail) === toAddress.length - toAddress.length) {
```
Comment (zgayjjf) [inline src/containers/WalletTransfer/WalletTransfer.vue:823]: Confusing.
A simple `/.bit$/.test(toAddress)` is just fine.

**Review by zgayjjf — CHANGES_REQUESTED** [review]

File: src/containers/WalletTransfer/WalletTransfer.vue [hunk 3]
```diff
@@ -898,6 +927,12 @@ export default {
     ensRule() {
       return this.selectedVerifier === ENS && this.ensError ? this.ensError : true
     },
+    bitRule() {
+      return this.selectedVerifier === BIT && this.bitError ? this.bitError : true
+    },
+    getToAddressComboboxItems() {
+      return !this.isBitMode ? this.contactList : this.multipleAddress
```
Comment (zgayjjf) [inline src/containers/WalletTransfer/WalletTransfer.vue:934]: Flip `?:` statement will be more clear and straightforward.
`this.isBitMode ? this.multipleAddress : this.contactList`

File: src/containers/WalletTransfer/WalletTransfer.vue [hunk 4]
```diff
@@ -909,7 +944,104 @@ export default {
...
+      this.contactSelected = input
+
+      // check tail of .bit begin
+      this.multipleAddress = []
+      this.bitError = ''
+      this.toAddress = ''
+      this.toEthAddress = ''
+      const tail = DOT_STRING + BIT
+      let addressCount = 0
+      if (input.indexOf(tail) > 0 && input.indexOf(tail) === input.length - tail.length) {
```
Comment (zgayjjf) [inline src/containers/WalletTransfer/WalletTransfer.vue:963]: `/.bit$/.test(input)` will be more clear and straghtforward.

File: src/containers/WalletTransfer/WalletTransfer.vue [hunk 5]
```diff
@@ -134,25 +134,39 @@
...
                       item-value="value"
                       aria-label="Recipient Address"
-                      :return-object="false"
+                      :return-object="getReturnObject()"
                       @input="contactChanged"
+                      @blur="checkContact"
+                      @update:search-input="listenInput"
                     >
                       <template v-if="apiStreamSupported" #append>
+                        <v-chip v-show="addressLogoUrl && isBitMode && toAddress && selectedVerifier === bitVerifier" class="address-chip">
```
Comment (zgayjjf) [inline src/containers/WalletTransfer/WalletTransfer.vue:151]: Seems like `selectedVerifier === bitVerifier` and `isBitMode` is reduntant.

File: src/containers/WalletTransfer/WalletTransfer.vue [hunk 6]
```diff
@@ -909,7 +944,104 @@ export default {
...
+        return
+      }
+      this.contactSelected = input
+
+      // check tail of .bit begin
+      this.multipleAddress = []
+      this.bitError = ''
+      this.toAddress = ''
+      this.toEthAddress = ''
+      const tail = DOT_STRING + BIT
```
Comment (zgayjjf) [inline src/containers/WalletTransfer/WalletTransfer.vue:961]: Seems like `this.bitTail` and `tail` here are the same things.

Comment (bigbigkent) [reply src/containers/WalletTransfer/WalletTransfer.vue]: done

File: src/containers/WalletTransfer/WalletTransfer.vue (hunk 4 above)
Comment (bigbigkent) [reply src/containers/WalletTransfer/WalletTransfer.vue]: done

**Review by zgayjjf — APPROVED** [review]

**Comment by himanshuchawla009** [comment]
Screenshot 2022-05-31 at 10 13 16 AM

can we add a fallback icon if user's bit icons doesn't found.

**Comment by himanshuchawla009** [comment]
Screenshot 2022-05-31 at 10 19 17 AM

ens domains and not working properly after adding this functionality. this address exists on ens but it is giving false error on ui.

Steps to reproduce:-

1. Enter a valid bit address
2. Switch the dropdown to ens
3. enter a valid ens address

Expected:- It should allow me to proceed with correct ens address

Current Result:- It is showing ens address as invalid.

**Review by zgayjjf — APPROVED** [review]

**Comment by bigbigkent** [comment]
> Screenshot 2022-05-31 at 10 13 16 AM
> 
> can we add a fallback icon if user's bit icons doesn't found.

done

**Comment by bigbigkent** [comment]
> Screenshot 2022-05-31 at 10 19 17 AM
> 
> ens domains and not working properly after adding this functionality. this address exists on ens but it is giving false error on ui.
> 
> Steps to reproduce:-
> 
> 1. Enter a valid bit address
> 2. Switch the dropdown to ens
> 3. enter a valid ens address
> 
> Expected:- It should allow me to proceed with correct ens address
> 
> Current Result:- It is showing ens address as invalid.

done

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**